src/request-data/resources.js

```diff
@@ -15,18 +15,21 @@ import { mergeDeepLeft } from 'ramda';
 import configDefaults from '../config';
 import { computeIfExists, hasVerbs, setupOption, transformForm } from './util';
 import { noargs } from '../util/util';
+import { _container } from './resource';
+import UserPreferences from './user-preferences/preferences';
 
 export default ({ i18n }, createResource) => {
```

Comment:
The first argument here, from which `i18n` is destructured, is the container. You can get `http` from there as well.